Dikke Ploaten/Services/Database.swift

```diff
@@ -7,55 +7,109 @@
-			let db = Firestore.firestore()
-			db.collection("users").document(user!.user.uid).setData(["username": username, "email": email]) { err in
-				// Error adding user to database
-				if let err = err {
-					print("Error adding document: \(err)")
-				}
-			}
-			
-			successHandler()
-		}
-	}
+    
+    let db = Firestore.firestore()
+    
+    // TODO: completionhandler
+    func addToDatabase(album: Album/*, completionHandler: @escaping () -> ()*/) {
+        // Data from object in JSON
+        let data = album.toJSON()
+        // Upload data to database
+        Firestore.firestore().collection("userPlaten").addDocument(data: data) { err in
+            if let err = err {
+                print("Error adding document: \(err)")
+                //completionHandler()
+            } else {
+                print("Document added with ID")
+            }
+        }
+    }
+    
+    func updateCollection(albums: [Album], completionHandler: @escaping (_ updatedCollection: [Album]) -> ()) {
+        var albums = albums
+        db.collection("userPlaten").limit(to: 1000).addSnapshotListener { querySnapshot, error in
+            guard let snapshot = querySnapshot else {
+                print("Error fetching snapshots: \(error!)")
+                return
+            }
+            for diff in snapshot.documentChanges {
+                if (diff.type == .added) {
+                    let album = try! Mapper<Album>().map(JSON: diff.document.data())
+                    album.id = diff.document.documentID
+                    if album.userID == Auth.auth().currentUser?.uid {
+                        albums.append(album)
+                    }
+                }
+                if (diff.type == .removed) {
+                    let album = try! Mapper<Album>().map(JSON: diff.document.data())
+                    album.id = diff.document.documentID
+                    if let index = albums.firstIndex(of: album) {
+                        albums.remove(at: index)
+                    }
+                }
+            }
+            completionHandler(albums)
+        }
+    }
+    
+    func getAlbumList(albums: [Album], completionHandler: @escaping (_ updatedCollection: [Album]) -> ()){
+        var albumList = albums
+        db.collection("platen").getDocuments() { (querySnapshot, err) in
+            if let err = err {
+                print("Error getting documents: \(err)")
+            } else {
+                for document in querySnapshot!.documents {
+                    print("\(document.documentID) => \(document.data())")
+                    let album = try! Mapper<Album>().map(JSON: document.data())
+                    album.id = document.documentID
+                    albumList.append(album)
+                }
+                completionHandler(albumList)
+            }
+        }
+    }
+    
+    func deleteAlbum(albums: [Album], indexPath: IndexPath ,completionHandler: @escaping (_ updatedCollection: [Album]) -> ()) {
```

Comment:
Graag geen indexPath meegeven, maar eenvoudigweg het id van het album dat je wil deleten. Ook geen idee waarom je albums hier meegeeft, er verandert toch niks aan de lijst, en je geeft die gewoonweg terug mee aan de completionHandler

Reply:
Zal ik aanpassen, ziet de completionhandler er dan gewoon zo () -> () uit?

Reply:
Wel ik zou die `(Error?) -> ()` maken, zodat je in je viewcontroller kan checken op de error en die kan gaan tonen. Hier ga je nu enkel bij success de completionhandler gaan aanroepen, wat een issue kan zijn als je bijvoorbeeld een loader wil tonen, dan gaat die loader nooit weggaan, aangezien je geen callback krijgt.